File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/BeregningsInformasjon.java
package no.nav.domain.pensjon.regler.beregning2011;

import no.nav.domain.pensjon.regler.Merknad;
import no.nav.domain.pensjon.regler.PenPerson;
import no.nav.domain.pensjon.regler.beregning.Sluttpoengtall;
import no.nav.domain.pensjon.regler.kode.BeregningMetodeTypeCti;
import no.nav.domain.pensjon.regler.kode.JustertPeriodeCti;
import no.nav.domain.pensjon.regler.kode.LandCti;
import no.nav.domain.pensjon.regler.kode.ResultatTypeCti;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class BeregningsInformasjon implements Serializable {

    //Fra IBeregningsInformasjon2011
    private double forholdstallUttak;
    private double forholdstall67;
    private double delingstallUttak;
    private double delingstall67;
    private Sluttpoengtall spt;
    private Sluttpoengtall opt;
    private Sluttpoengtall ypt;
    private boolean grunnpensjonAvkortet = false;
    private List<Merknad> gpAvkortingsArsakList = new ArrayList<Merknad>();
    private boolean mottarMinstePensjonsniva;
    private String minstepensjonArsak;
    private boolean rettPaGjenlevenderett;
    private boolean gjenlevenderettAnvendt;
    private boolean avdodesTilleggspensjonBrukt;
    private boolean avdodesTrygdetidBrukt;
    private boolean ungUfor;
    private boolean ungUforAnvendt;
    private boolean yrkesskadeRegistrert;
    private boolean yrkesskadeAnvendt;
    private int yrkesskadegrad;

    // Fra IBeregningsInformasjon
    private PenPerson penPerson;
    private BeregningMetodeTypeCti beregningsMetode;
    private boolean eksport;
    private ResultatTypeCti resultatType;
    private List<TapendeBeregningsmetode> tapendeBeregningsmetodeListe = new ArrayList<TapendeBeregningsmetode>();
    private Integer trygdetid;
    private int tt_anv;
    private LandCti vurdertBosattland;
    private boolean ensligPensjonInstOpph;
    private JustertPeriodeCti instOppholdType;
    private boolean instOpphAnvendt;
    // Avdødes tilleggspensjon
    private double tp;
    private int ttBeregnetForGrunnlagsrolle;

    private boolean ungUforGarantiFrafalt;


}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/BeregningsInformasjon.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.Merknad
import no.nav.pensjon.regler.domain.PenPerson
import no.nav.pensjon.regler.domain.beregning.Sluttpoengtall
import no.nav.pensjon.regler.domain.enum.BeregningsmetodeEnum
import no.nav.pensjon.regler.domain.enum.JustertPeriodeEnum
import no.nav.pensjon.regler.domain.enum.LandkodeEnum
import no.nav.pensjon.regler.domain.enum.ResultattypeEnum
import java.io.Serializable

class BeregningsInformasjon : Serializable {
    var forholdstallUttak = 0.0
    var forholdstall67 = 0.0
    var delingstallUttak = 0.0
    var delingstall67 = 0.0
    var spt: Sluttpoengtall? = null
    var opt: Sluttpoengtall? = null
    var ypt: Sluttpoengtall? = null
    var grunnpensjonAvkortet = false
    var merknadListe: List<Merknad> = mutableListOf()
    var mottarMinstePensjonsniva = false
    var minstepensjonArsak: String? = null
    var rettPaGjenlevenderett = false
    var gjenlevenderettAnvendt = false
    var avdodesTilleggspensjonBrukt = false
    var avdodesTrygdetidBrukt = false
    var ungUfor = false
    var ungUforAnvendt = false
    var yrkesskadeRegistrert = false
    var yrkesskadeAnvendt = false
    var yrkesskadegrad = 0
    var penPerson: PenPerson? = null
    var beregningsMetodeEnum: BeregningsmetodeEnum? = null
    var eksport = false
    var resultatTypeEnum: ResultattypeEnum? = null
    var tapendeBeregningsmetodeListe: List<TapendeBeregningsmetode> = mutableListOf()
    var trygdetid: Int? = null
    var tt_anv = 0
    var vurdertBosattlandEnum: LandkodeEnum? = null
    var ensligPensjonInstOpph = false
    var instOppholdTypeEnum: JustertPeriodeEnum? = null
    var instOpphAnvendt = false
    var tp = 0.0
    var ttBeregnetForGrunnlagsrolle = 0
    var ungUforGarantiFrafalt = false
}